project_generator/tools/gccarm.py

```diff
@@ -113,6 +113,14 @@ def process_data_for_makefile(self, project_data):
         project_data['toolchain'] = 'arm-none-eabi-'
         project_data['toolchain_bin_path'] = self.env_settings.get_env_settings('gcc')
 
+        linker_file = project_data['linker_file']
```

Comment:
https://github.com/project-generator/project_generator/blob/master/project_generator/project.py#L492

this one does it generically (no tool involved). You cna improve that check there, but should not be in the tools?